server/logger.go

```diff
@@ -0,0 +1,32 @@
+package server
+
+import (
+	"fmt"
+	"log"
+)
+
+type Logger interface {
+	log(format string, v ...interface{})
+	logError(format string, v ...interface{})
+}
+
+type Log struct {
+	uid, ip, origin string
+}
+
+func NewLogger(uid, ip, origin string) Logger {
+	return &Log{
+		uid:    uid,
+		ip:     ip,
+		origin: origin,
+	}
+}
+func (l *Log) log(format string, v ...interface{}) {
+	prefix := fmt.Sprintf("[%s] [%s] [%s] ", l.uid, l.ip, l.origin)
```

Comment:
I fear the logs will become too verbose with IP and Origin added to each entry. Perhaps we should leave it at only uid

Reply:
Updated.